### PR TITLE
Debug Info

### PR DESCRIPTION
**Pull request by CodaFi**

### CODE REVIEW

**Comment by harlanhaskins** [comment]
This looks good so far! One small nit: I think `Metadata` should be one word.

**Comment by CodaFi** [comment]
The operative verb in all of the DI bindings is "create" instead of "build".  Should we match LLVM or our IRBuilder?

**Comment by harlanhaskins** [comment]
So I think the difference is that IRBuilder builds instructions in a scope, while DIBuilder creates types/relationships/metadata that are attached to instructions. So I think `create` works for that

File: Sources/LLVM/DIBuilder.swift [hunk 1]
```diff
@@ -362,6 +362,30 @@ public enum DWARFTypeEncoding {
   case ucs
   
   case ascii
+
+  static let typeEncodingMapping: [DWARFTypeEncoding: LLVMDWARFTypeEncoding] = [
```
Comment (segiddins) [inline Sources/LLVM/DIBuilder.swift]: should this maybe be `private`?

File: Sources/LLVM/DIBuilder.swift [hunk 2]
```diff
@@ -0,0 +1,1093 @@
...
+  /// Create a permanent forward-declared type.
+  public func createForwardDecl(in scope: Scope, named name: String, in file: FileMetadata, at line: Int, tag: Int) -> DIType {
+    guard let fd = LLVMDIBuilderCreateForwardDecl(self.llvm, UInt32(tag), name, scope.llvm, file.llvm, UInt32(line)) else {
+      fatalError("Failed to allocate metadata for a forward declaration")
+    }
+    return AnyDIType(llvm: fd)
+  }
+
+  /// Create a new descriptor for a parameter variable.
+  public func createParameterVariable(in scope: Scope, named name: String, at index: UInt32, in file: FileMetadata, at line: Int, type: DIType, _ alwaysPreserve: Bool = true, flags: DebugInfoFlags = .zero) -> VariableMetadata {
```
Comment (harlanhaskins) [inline Sources/LLVM/DIBuilder.swift]: Let's make all the indices Ints. Swift stdlib did this too.

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**